packages/lib/utils/recall.ts

```diff
@@ -0,0 +1,100 @@
+import { TSurvey, TSurveyQuestion } from "@formbricks/types/surveys";
+
+export function extractId(text: string) {
+  const pattern = /recall:([A-Za-z0-9]+)/;
+  const match = text.match(pattern);
+  if (match && match[1]) {
+    return match[1];
+  } else {
+    return null;
+  }
+}
+
+export function extractIds(text: string) {
```

Comment:
GPT note:

To optimize the `extractIds` function and eliminate the redundant use of regular expressions, you can utilize capturing groups in your initial regular expression. This way, you can directly extract the necessary IDs from the initial match without needing to reapply a regular expression within the `map` function.

Here is how you can refactor the `extractIds` function:

```javascript
export function extractIds(text: string) {
  const pattern = /recall:([A-Za-z0-9]+)/g;
  const matches = [...text.matchAll(pattern)];
  return matches.map(match => match[1]).filter(id => id !== null);
}
```

Explanation:
- `text.matchAll(pattern)` is used instead of `text.match(pattern)`. This method returns an iterator of all matches, including capturing groups.
- By spreading the iterator into an array with `[...text.matchAll(pattern)]`, you get an array of match objects. Each match object includes the entire matched string and the captured groups.
- `match[1]` directly accesses the first captured group (the ID part in your case) of each match. This eliminates the need for a second regular expression match inside the `map`.
- The `filter(id => id !== null)` part is to ensure that only valid IDs are included in the final array.

This refactoring makes the function more efficient by reducing the number of regular expression operations and simplifying the logic.